datastream-server/src/test/java/com/linkedin/datastream/server/TestFileBasedPartitionThroughputProvider.java

```diff
@@ -0,0 +1,34 @@
+/**
+ *  Copyright 2021 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.server;
+
+import java.util.HashMap;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import com.linkedin.datastream.server.providers.FileBasedPartitionThroughputProvider;
+
+
+/**
+ * Tests for {@link FileBasedPartitionThroughputProvider}
+ */
+public class TestFileBasedPartitionThroughputProvider {
+
```

Comment:
For the open source `File` based solution, you may want to add more negative tests. like cluster not present, file not present.